nvme.c

```diff
@@ -5317,6 +5398,629 @@ static int show_registers(int argc, char **argv, struct command *cmd, struct plu
+		 OPT_FLAG("human-readable", 'H', &cfg.human_readable, human_readable),
+		 OPT_FLAG("cap",            'c', &cfg.cap,            cap),
+		 OPT_FLAG("vs",             'V', &cfg.vs,             vs),
+		 OPT_FLAG("cmbloc",         'm', &cfg.cmbloc,         cmbloc),
+		 OPT_FLAG("cmbsz",          'M', &cfg.cmbsz,          cmbsz),
+		 OPT_FLAG("bpinfo",         'b', &cfg.bpinfo,         bpinfo),
+		 OPT_FLAG("cmbsts",         'S', &cfg.cmbsts,         cmbsts),
+		 OPT_FLAG("cmbebs",         'E', &cfg.cmbebs,         cmbebs),
+		 OPT_FLAG("cmbswtp",        'W', &cfg.cmbswtp,        cmbswtp),
+		 OPT_FLAG("crto",           'r', &cfg.crto,           crto),
+		 OPT_FLAG("pmrcap",         'P', &cfg.pmrcap,         pmrcap),
+		 OPT_FLAG("pmrsts",         't', &cfg.pmrsts,         pmrsts),
+		 OPT_FLAG("pmrebs",         'e', &cfg.pmrebs,         pmrebs),
+		 OPT_FLAG("pmrswtp",        'w', &cfg.pmrswtp,        pmrswtp),
+		 OPT_FLAG("intms",          'i', &cfg.intms,          intms),
+		 OPT_FLAG("intmc",          'I', &cfg.intmc,          intmc),
+		 OPT_FLAG("cc",             'C', &cfg.cc,             cc),
+		 OPT_FLAG("csts",           'T', &cfg.csts,           csts),
+		 OPT_FLAG("nssr",           'n', &cfg.nssr,           nssr),
+		 OPT_FLAG("aqa",            'a', &cfg.aqa,            aqa),
+		 OPT_FLAG("asq",            'A', &cfg.asq,            asq),
+		 OPT_FLAG("acq",            'q', &cfg.acq,            acq),
+		 OPT_FLAG("bprsel",         'B', &cfg.bprsel,         bprsel),
+		 OPT_FLAG("bpmbl",          'p', &cfg.bpmbl,          bpmbl),
+		 OPT_FLAG("cmbmsc",         's', &cfg.cmbmsc,         cmbmsc),
+		 OPT_FLAG("nssd",           'N', &cfg.nssd,           nssd),
+		 OPT_FLAG("pmrctl",         'R', &cfg.pmrctl,         pmrctl),
+		 OPT_FLAG("pmrmscl",        'l', &cfg.pmrmscl,        pmrmscl),
+		 OPT_FLAG("pmrmscu",        'u', &cfg.pmrmscu,        pmrmscu));
+
+	err = parse_and_open(&dev, argc, argv, desc, opts);
+	if (err)
+		return err;
+
+	err = validate_output_format(output_format_val, &flags);
+	if (err < 0) {
+		nvme_show_error("Invalid output format");
+		return err;
+	}
+
+	if (cfg.human_readable)
+		flags |= VERBOSE;
+
+	r = nvme_scan(NULL);
+	bar = mmap_registers(r, dev, false);
+	if (!bar) {
+		err = get_register_property(dev_fd(dev), &bar, NVME_REG_PMRMSCU);
+		if (err)
+			return err;
+		fabrics = true;
+	}
+
+	if (!get_register_offset(bar, fabrics, &cfg, flags)) {
+		nvme_show_error("offset required param");
+		err = -EINVAL;
+	}
+
+	if (fabrics)
+		free(bar);
+	else
+		munmap(bar, getpagesize());
+
+	return err;
+}
+
+static void nvme_set_register(void *bar, int offset, uint64_t value, bool mmio32)
+{
+	if (nvme_is_64bit_reg(offset))
+		mmio_write64(bar + offset, value, mmio32);
+	else
+		mmio_write32(bar + offset, value);
+
+	printf("set-register: %02x (%s), value: %"PRIx64"\n", offset,
+	       nvme_register_to_string(offset), value);
+}
+
+static void set_register_value(void *bar, struct argconfig_commandline_options *opts,
```

Comment:
This function consists of a `switch` over `offset`, but each caller specifies a constant `offset` value. I would restructure the code to inline each switch case in the caller, which would avoid the unnecessary branching for every `offset` value.

Reply:
Okay will do consider the advice later anyway understootd the current set-reg command needed to refactor. By the way can you show me a simple example for the suggetion to mare sure?

Reply:
As mentioned by the comment https://github.com/linux-nvme/nvme-cli/pull/2188#issuecomment-1951325296 just refactored the set-reg command by using for loops instead of the repeated funciton call and also changed the functions as inline funcion as mentioned.